neon_core/gui/resting_screen.py

```diff
@@ -0,0 +1,226 @@
+# # NEON AI (TM) SOFTWARE, Software Development Kit & Application Development System
+# # All trademark and other rights reserved by their respective owners
+# # Copyright 2008-2021 Neongecko.com Inc.
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+# 1. Redistributions of source code must retain the above copyright notice,
+#    this list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+# 3. Neither the name of the copyright holder nor the names of its
+#    contributors may be used to endorse or promote products derived from this
+#    software without specific prior written permission.
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS  BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
+# OR PROFITS;  OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+# LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE,  EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import time
+
+from threading import Lock
+from neon_utils.log_utils import LOG
+from mycroft_bus_client import Message, MessageBusClient
+from mycroft.skills.mycroft_skill import MycroftSkill
+
+
+def compare_origin(m1, m2):
+    origin1 = m1.data["__from"] if isinstance(m1, Message) else m1
+    origin2 = m2.data["__from"] if isinstance(m2, Message) else m2
+    return origin1 == origin2
+
+
+class RestingScreen:
+    """
+    Implementation of resting screens.
+    This class handles registration and override of resting screens,
+    encapsulating the system.
+    """
+
+    def __init__(self):
+        skill = MycroftSkill()
+        bus = MessageBusClient()
+        bus.run_in_thread()
+        skill.bind(bus)
+        self.bus = skill.bus
+        self.gui = skill.gui
```

Comment:
better idea than comment above, we can skip the need to for a dummy MycroftSkill class if we use https://github.com/OpenVoiceOS/ovos_utils/blob/dev/ovos_utils/gui.py#L382

this is the same as the SkillGUI class but does not require a skill object, it is what i use in the ovos-common-play framework

in this case we init self.gui with only a skill_id string + the bus object

Reply:
We would still need the event scheduling methods from MycroftSkill though (could be implemented separately, but I think scope of work is enough for a follow-up PR)